docs/overview.yaml

```diff
@@ -61,7 +61,7 @@ General algebra:
     ideal of a commutative ring: 'ideal'
     quotient ring: 'ideal.has_quotient'
     prime ideal: 'ideal.is_prime'
-    maximal ideal: 'ideal.is_maximal'
+    maximal ideal (as maximal submodule): 'submodule.is_maximal'
```

Comment:
I think it's probably clearer to keep "maximal ideal" there. People clicking this link may be confused for a few seconds but they will very quickly read the docstring.

Reply:
I agree with keeping "maximal ideal" in docs since it's in "Ideals and quotients" sections and shorter is better, so I removed `(as maximal submodule)`.